src/plugins/data_views/common/data_views/data_view.ts

```diff
@@ -290,11 +298,11 @@ export class DataView implements IIndexPattern {
     return [...this.fields.getAll().filter((field) => field.scripted)];
   }
 
-  isTimeBased(): boolean {
+  isTimeBased(): this is TimeBasedDataView {
```

Comment:
:information_source: This way we get better TS support. Calling that method now, will cause the compiler to understand, that if that returns `true` `timeFieldName` must be a `string` and can't be `undefined` anymore.